### PR TITLE
Modify ResourceInterceptor interface

### PR DESCRIPTION
**Pull request by suleymanakbas91**
See https://github.com/kyma-incubator/reconciler/issues/524 (more specifically https://github.com/kyma-incubator/reconciler/issues/524#issuecomment-988083858)

### CODE REVIEW

File: pkg/reconciler/kubernetes/adapter_test.go [hunk 1]
```diff
@@ -55,13 +56,16 @@ var expectedResourcesWithNs = []*Resource{
...
 type testInterceptor struct {
-	result InterceptionResult
-	err    error
+	err error
 }
 
-func (i *testInterceptor) Intercept(resource *unstructured.Unstructured, _ string) (InterceptionResult, error) {
-	resource.SetLabels(expectedLabels)
-	return i.result, i.err
+func (i *testInterceptor) Intercept(resources map[string][]*unstructured.Unstructured, _ string) error {
```
Comment (tobiscr) [inline pkg/reconciler/kubernetes/adapter_test.go:62]: Maybe for the later PR: could we make the map to an object which offers some convenient looping and getter function.

E.g.

     type Resources map[string][]*unstructured.Unstructured

     func (r resources) Visit(callback func(kind string, u *unstructured.Unstructured) error) error {
     	for kind := range r {
     		for _, resource := range r[kind] {
     			if err := callback(kind, resource); err != nil {
     				return err
     			}
     	     	}
     	}
     	return nil
     }

    func (r resources) Get(kind string) []*unstructured.Unstructured {
      	return r[kind]
     }

**Review by tobiscr — DISMISSED** [review]

**Review by tobiscr — APPROVED** [review]

File: pkg/reconciler/kubernetes/adapter_test.go (hunk 1 above)
Comment (suleymanakbas91) [reply pkg/reconciler/kubernetes/adapter_test.go]: Done

**Review by tobiscr — APPROVED** [review]

**Pull request merged**